### PR TITLE
add version.h generated from the numbers in CMakeLists.txt

### LINKED ISSUE #874: Sirf version to be available at C++ level
With #872 version strings have been added at Python and MATLAB level, however this should be available in C++.

### PR DESCRIPTION
**Pull request by paskino**
closes #874

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -188,12 +188,16 @@ endif(DISABLE_Matlab)
 
 ENABLE_TESTING()
 
+configure_file("cmake/version.h.in" "${CMAKE_CURRENT_BINARY_DIR}/cmake/version.h")
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cmake/version.h" DESTINATION "${CMAKE_INSTALL_PREFIX}/include/sirf/common/")
```
Comment (KrisThielemans) [inline CMakeLists.txt:192]:
```suggestion
configure_file("cmake/version.h.in" "${CMAKE_CURRENT_BINARY_DIR}/cmake/include/sirf/common/version.h")
install(FILES "${CMAKE_CURRENT_BINARY_DIR}/cmake/include/sirf/common/version.h" DESTINATION "${CMAKE_INSTALL_PREFIX}/include/sirf/common/")
```

File: src/common/tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+add_executable(test_version test_version.cpp)
+target_include_directories(test_version PUBLIC ${CMAKE_BINARY_DIR}/cmake)
```
Comment (KrisThielemans) [inline src/common/tests/CMakeLists.txt:2]:
```suggestion
target_include_directories(test_version PUBLIC ${CMAKE_BINARY_DIR}/cmake/include)
```

File: src/common/tests/test_version.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include "version.h"
```
Comment (KrisThielemans) [inline src/common/tests/test_version.cpp:1]:
```suggestion
#include "sirf/common/version.h"
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
could add another `git_version.h` file from `git describe` as discussed in #874, or leave that for later.

**Comment by paskino** [comment]
I wonder what would be the use of `git_version.h`. It'd add confusion, IMHO.

**Pull request merged**